tests/pyccel/test_pyccel.py

```diff
@@ -247,7 +248,7 @@ def teardown(path_dir = None):
             shutil.rmtree( file_name )
         elif not os.path.isfile(file_name):
             teardown(file_name)
-        elif not f.endswith(".py"):
+        elif not f.endswith(".py") and not f.endswith(".pyh") and not f.endswith(".pyccel"):
```

Comment:
Shouldn't we delete the `.pyccel` file after the test to be sure that we have a clean environment? That way we don't need to test if the integer case has a float header (probably harder to test)

Reply:
Equally if you write the `.pyh` file each time then can't it be deleted after the test?

Reply:
lol the whole purpose of the test is to keep the `.pyccel` file

Reply:
if the integer case has a float header, then `int(lang_out)` will raise an error, because you are doing something like
`int('1.')`

Reply:
This is called after the test. Obviously the `.pyccel` file needs to be kept during the test between the `int` case and the `float` case, but should it be kept between the `float` case and a potential second run starting with the `int` case?

Reply:
it should be fine

Reply:
> if the integer case has a float header, then int(lang_out) will raise an error, because you are doing something like
int('1.')

As long as it is printed in the correct format then yes. This seems to be the case for both c and fortran